src/ca65/main.c

```diff
@@ -1040,7 +1048,14 @@ int main (int argc, char* argv [])
     /* If no CPU given, use the default CPU for the target */
     if (GetCPU () == CPU_UNKNOWN) {
         if (Target != TGT_UNKNOWN) {
-            SetCPU (GetTargetProperties (Target)->DefaultCPU);
+            int cpu;
+
+            cpu = GetTargetProperties (Target)->DefaultCPU;
+
+            if (cpu == CPU_UNKNOWN) {
+              Error ("CPU of target is unknown");
+            }
+            SetCPU (cpu);
```

Comment:
That test isn't needed.  None of the `DefaultCPU` entries is allowed to have `CPU_UNKNOWN` -- even the `none` target has a valid CPU.